### PR TITLE
cmd/utils: bugfix where database is opened multiple times

### LINKED ISSUE #2428: Export chain fails
#### System information

Geth version: 1.4 develop
OS & Version: Windows/Linux/OSX
Commit hash : 7e02105672cda92889a78db864a5701d78f45eb2

#### Expected behaviour
`geth export <filename>` dumps chain from datadir to filename

#### Actual behaviour
```
bas@zitroen git:develop :-) ./build/bin/geth export /tmp/chain.export
I0406 12:47:19.444898 ethdb/database.go:82] Alloted 128MB cache and 1024 file handles to /home/bas/.ethereum/chaindata
I0406 12:47:19.677214 ethdb/database.go:82] Alloted 128MB cache and 1024 file handles to /home/bas/.ethereum/chaindata
Fatal: Could not open database: resource temporarily unavailable
```

#### Steps to reproduce the behaviour
`geth export <filename>`

#### Problem 
Relevant line.
https://github.com/ethereum/go-ethereum/blob/7e02105672cda92889a78db864a5701d78f45eb2/cmd/utils/flags.go#L832

### PR DESCRIPTION
**Pull request by bas-vk**
This PR addresses an issue with exporting and importing the chain to/from file. Geth tries to open the database multiple times which would lead to a crash.

Fixes #2428 

@obscuren, PTAL.

I'm not sure about the intentions of [this](https://github.com/ethereum/go-ethereum/blob/7e02105672cda92889a78db864a5701d78f45eb2/cmd/utils/flags.go#L826) line. If the olympic flag is set then the testnet genesis block is written? It was added later so there seems to be reason for it?

### CODE REVIEW

**Comment by robotally** [comment]
| Vote | Count | Reviewers |
| :---: | :---: | :---: |
| :+1: | 0 |  |
| :-1: | 0 |  |

_Updated: Thu Apr 7 09:39:06 UTC 2016_

**Comment by fjl** [comment]
The commit message says "core: ..." but the fix is in cmd/utils. It looks good otherwise.

**Comment by obscuren** [comment]
I've no idea tbh @bas-vk 

In any case :+1:

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -776,16 +776,20 @@ func SetupNetwork(ctx *cli.Context) {
...
-		genesis = core.GetBlock(db, core.GetCanonicalHash(db, 0))
-	)
+	db := MakeChainDatabase(ctx)
 	defer db.Close()
 
+	return MustMakeChainConfigFromDb(ctx, db)
+}
+
+// MustMakeChainConfigFromDb reads the chain configuration from the given database.
+func MustMakeChainConfigFromDb(ctx *cli.Context, db ethdb.Database) *core.ChainConfig {
```
Comment (karalabe) [inline cmd/utils/flags.go]: Do we use `ctx` for anything? If not, please drop it.

Comment (bas-vk) [inline cmd/utils/flags.go]: The returned `ChainConfig` contains the homestead block number. `ctx` is used to determine if we are on the test or live net which have different homestead block numbers.

Comment (karalabe) [inline cmd/utils/flags.go]: Ah, sorry, indeed, I didn't see that line.

**Comment by karalabe** [comment]
LGTM :+1:

**Pull request merged**